doc/org/build/gameplay.md

```diff
@@ -0,0 +1,722 @@
+- [Intro](#org71ab195)
+- [Entities, components, systems.](#org739a535)
+  - [`bitECS`](#org0bfe8ab)
+  - [Disclaimer](#orgff3efea)
+- [Writing systems](#org751fbe1)
+  - [Systems are functions](#org775dcd0)
+  - [The game loop](#orga589afa)
+  - [Queries are useful](#org1d5d556)
+  - [Replacing `async` functions with `coroutines`](#org2233e62)
+- [Writing components](#org7331485)
+  - [Defining components](#org18a6d0b)
+  - [Data types](#org3db6503)
+  - [Avoid holding references](#org589f3d9)
+  - [Entity ID&rsquo;s are recycled](#org31d40f6)
+  - [String data](#orgdf10967)
+  - [Flags](#org70bd5dd)
+  - [Tag components](#org58d595d)
+  - [The escape hatch](#orga5547d0)
+  - [Associating entities with `Object3D` s](#orgf29a631)
+  - [Avoid duplicating state](#orgfe9fb09)
+- [Adding entities](#orgbad6d63)
+  - [Entity basics](#org9d285be)
+  - [Creating `EntityDef` s](#orgdd790f5)
+  - [Creating model files](#org156c909)
+  - [Entity creation is synchronous](#org2319bae)
+  - [Inflation](#org628683b)
+    - [What does `renderAsEntity` do?](#org20a38aa)
+    - [`Inflator` s](#org357d40d)
+    - [Default inflators](#orga810c47)
+    - [Associating `Object3D` s (`eid2obj`)](#org4f9fcd3)
+    - [Loading model files](#org7a41123)
+    - [Common inflators, `jsxInflators`, and `gltfInflators`](#org8ca2b33)
+    - [Entity `Ref` s and `__mhc_link_type` : `"node"`](#orgfcfa191)
+    - [Associating `Material` s (`eid2mat`)](#org7fe068b)
+- [Custom clients and addons](#org1a8110a)
+  - [Addons are not ready yet (February 2023)](#org2b06d43)
+  - [Creating an add-on](#org35960ac)
+  - [`preload`](#orge79e4e1)
+  - [Inserting prefabs](#org1d14013)
+  - [Inserting inflators](#org86ae847)
+  - [Inserting system calls](#org354b138)
+  - [Handling interactions](#org306c661)
+  - [Handling networking](#org19b999f)
+
+\#+TITLE Core Concepts for Gameplay Code
+
+Core Concepts for Gameplay Code
+
+
+<a id="org71ab195"></a>
+
+# Intro
+
+This document gives an overview of the core concepts for writing gameplay code in the Hubs client.
+
+
+<a id="org739a535"></a>
+
+# Entities, components, systems.
+
+ECS became a popular topic in recent years.
+
+-   Unity&rsquo;s `DOTS` emphasizes data-oriented design for speed and control, separates behavior from data, and helps developers build multi-threaded game loops.
+-   Supermedium&rsquo;s `A-Frame` emphasizes ease of use and a low barrier to entry, exposes three.js through familiar HTML, and enables rapid prototyping with many built-in components and hundreds more from the community.
+
+Originally built with `A-Frame`, Hubs switched to `bitECS` and using `three.js` directly. Motivation, goals, and non-goals about the transition can be found in this PR from June, 2022. <https://github.com/mozilla/hubs/pull/5536>
+
+
+<a id="org0bfe8ab"></a>
+
+## `bitECS`
+
```

Comment:
Maybe better to first mention that bitECS is ECS framework written in TypeScript (JavaScript) and add a link to bitECS project.